### PR TITLE
fix: DH-22949: race condition/work duplication in UpdateBy caching

### PR DESCRIPTION
**Pull request by lbooker42**
Fixes DH-22949 by preventing duplicate work (and resulting race/NPE) when UpdateBy operators have duplicate input columns (e.g., value column equals weight column for rolling weighted average), and adds a regression test intended to reproduce the issue.

**Changes:**
- De-duplicate operator input source indices before caching to avoid parallel caching tasks operating on the same source index.
- Add a DH-22949 reproduction/regression test for bucketed `RollingWAvg` with duplicate inputs.

### CODE REVIEW

File: engine/table/src/test/java/io/deephaven/engine/table/impl/updateby/TestRollingWAvg.java [hunk 1]
```diff
@@ -709,6 +711,38 @@ public void testValueColumnEqualsWeightColumn() {
...
+                    System.nanoTime() + java.util.concurrent.TimeUnit.SECONDS.toNanos(SECONDS_TO_RUN);
+            final MutableInt iteration = new MutableInt(0);
+            do {
+                System.out.println("iteration " + iteration.getAndIncrement() + ", "
+                        + (deadlineNanos - System.nanoTime()) / 1_000_000_000.0 + "s remaining");
+                final Table result = table.updateBy(
+                        UpdateByOperation.RollingWAvg(10, "x", "rwx = x", "rwy = y"), "Sym");
+                Assert.eq(result.size(), "result.size()", table.size());
+            } while (System.nanoTime() < deadlineNanos);
+        }
```
[the remark this answers is not in the document]
Comment (lbooker42) [reply engine/table/src/test/java/io/deephaven/engine/table/impl/updateby/TestRollingWAvg.java]: Good point on eliminating the println

File: engine/table/src/test/java/io/deephaven/engine/table/impl/updateby/TestRollingWAvg.java [hunk 2]
```diff
@@ -709,6 +711,38 @@ public void testValueColumnEqualsWeightColumn() {
...
+        final QueryTable table = createTestTable(100, true, false, false, 0xABCDEF,
+                new String[] {"x", "y"},
+                new TestDataGenerator[] {
+                        new DoubleGenerator(10.1, 20.1, 0.0),
+                        new DoubleGenerator(30.1, 40.1, 0.0)
+                }).t;
+
+        // Without the fix, local testing failed 20/20 runs with NPE with maximum of 5 seconds before failure.
+        // 30 seconds anticipated to be sufficient to repro even on slower hardware (without delaying overall testing)
+        final int SECONDS_TO_RUN = 30;
```
Comment (cpwright) [inline engine/table/src/test/java/io/deephaven/engine/table/impl/updateby/TestRollingWAvg.java]: Let's fix this at 10.  How many iterations is that?  We might want to follow co-pilots advice and cut out the println.

Comment (lbooker42) [reply engine/table/src/test/java/io/deephaven/engine/table/impl/updateby/TestRollingWAvg.java]: Sounds good. 
With println, 24220 iterations in 10s,
w/o println, 31833 iterations

**Review by cpwright — APPROVED** [review]
